SWIPs/swip-honey_to_money.md

```diff
@@ -7,51 +7,66 @@ Type: Standards track
+```typescript 
+honeyPrices: { 
+ TTL,
+ prices: [0: {validFrom, honeyPrice}, n: {validFrom, honeyPrice}]
+}
+```
+* The applied price is the `price` where `validFrom` is in the most recent past. 
+* Upon receipt of a payment or intent to send a payment, nodes will check their local cache on the existence of a non-expired `honeyPrice` object. If this exists, the oracle will not be queried, if it does not exist, the oracle will be queried and the answer stored in the local cache of the node for `TTL` time. 
+* The number of payments received/sent, `TTL` of `honeyPrices`, as well as the costs to query the oracle (computational overload/time delay), should determine whether nodes actively keep their local cache up-to-date with `honeyPrices` or whether they will passively request a new `honeyPrices` object when needed.
 
 ## Rationale
-We need a way for prices in the Swarm to change over time. However, due to the nature of Swarm - specifically independent connections between pairs of nodes each with their individual accounting and payments - there is no clear price discovery mechanism. Furthermore, since Swarm nodes are expected to be conduits for both payments and data (buy from one peer and sell to the other) it is necessary for nodes to coordinate price changes with all their peers. In short, we need a mechanism for network-wide changes in pricing that the peer-to-peer routing and accounting does not provide.
-By externalising the price to an oracle, we can completely decouple Swarm's core functions from price discovery. By building support for price oracles, we can coordinate price changes across the network, and set the stage for future experimentation with various mechanisms for setting prices, from DAOs, to bonding curves, to fixed prices. This SWIP makes no recommendation for the eventual pricing mechanism, it only prepares the network for adopting them in future.
+<!--The rationale fleshes out the specification by describing what motivated the design and why particular design decisions were made. It should describe alternate designs that were considered and related work, e.g. how the feature is supported in other languages. The rationale may also provide evidence of consensus within the community, and should discuss important objections or concerns raised during discussion.-->
+### General rationale for choosing oracles
+There exist a need for the price of Swarm to change over time. However, due to the nature of Swarm, there is a need for this price to be decided network-wide, instead of peer-to-peer. Specifically:
+- The requirement on connectivity in the local bin and far-away bins means that nodes *must agree* with all nodes in their local bin and at least one node in a far-away bin on the price.
+- Zero marginal costs of serving data with perfect competition means that when nodes would engage in peer-to-peer price discovery, the price will tend to zero and a sub-optimal equilibrium of near-zero prices will be reached.
```

Comment:
again, this is not apparent to me. can you explain why this is, and why this matters in the context of prices being decided network-wide instead of p2p?

Reply:
Sure, but be ready to dive in a bit into economic theory (which all, in fact, is very logical):

**Perfect competition**
Is a market where there are numerous sellers who sell a nearly identical or indistinguishable product. The best example is in a market for raw material. I.e: one tonne of Grade A oak wood, is identical to another batch of Grade A oak wood. 
See: [Wikipedia - Idealizing conditions of perfect competition](https://en.wikipedia.org/wiki/Perfect_competition)

**Marginal-cost = marginal revenue**
This is the equation which all economics students learn in their first course. It holds true in a market of perfect competition and basically means that the price (marginal revenue) in the market for one extra good is equal to the cost of producing this good. Think about the situation of the oak wood producers. Here, if one (out of the many) oak-wood producers quote a price which is higher than the price of another producer, nobody would buy his oak (the quality is all the same). In equilibrium, this means that the producer capable of producing the oak for the lowest cost without going bankrupt himself will be able to sell his oak. Thus: the price for one extra unit of oak in the market equals the cost of producing this extra unit. Any sunk-costs (costs which are made already) are not included in the marginal costs as these are already lost anyways (by all producers).
[Marginal Cost Profit Maximization Strategy](https://courses.lumenlearning.com/boundless-economics/chapter/production-decisions-in-perfect-competition/)

**Applied to SWARM** 
The good we sell in Swarm are *chunks*. For one chunk, it is expected that many producers hold this chunk and the same chunk is always equal in quality to another chunk of the same kind. If Swarm is a market of perfect competition (and I would argue it comes very close), the price of a chunk is expected to go to the marginal costs of producing this extra chunk, which equals the bandwidth and electricity consumed for sending this chunk (almost 0).

As this situation is not desirable for anybody (nobody will be incentivized to boot up a Swarm node), there must be some sort of system where nodes can come to an agreement which is more efficient. What this is, that is the question, but anyway, it should include at least a little profit for Swarm operators, to cover their sunk-costs (hard-drive/pc/time invested).

Reply:
wonderful explanation! thanks.